packages/block-library/src/button/style.scss

```diff
@@ -50,57 +50,44 @@ $blocks-block__margin: 0.5em;
 	}
 
 	&.wp-block-button__width-25 {
-		width: calc(25% - #{ $blocks-block__margin });
+		width: calc(25% - ( var(--wp--style--block-gap, $blocks-block__margin) ));
 	}
 
 	&.wp-block-button__width-50 {
-		width: calc(50% - #{ $blocks-block__margin });
+		width: calc(50% - ( var(--wp--style--block-gap, $blocks-block__margin) ));
 	}
 
 	&.wp-block-button__width-75 {
-		width: calc(75% - #{ $blocks-block__margin });
+		width: calc(75% - ( var(--wp--style--block-gap, $blocks-block__margin) ));
 	}
 
 	&.wp-block-button__width-100 {
-		width: calc(100% - #{ $blocks-block__margin });
+		width: calc(100% - ( var(--wp--style--block-gap, $blocks-block__margin) ));
 
 		&:only-child {
-			margin-right: 0;
 			width: 100%;
 		}
 	}
 }
 
 // If the browser supports column-gap, use that instead of the margins above.
-@supports ( column-gap: #{ $blocks-block__margin } ) {
-	.wp-block-buttons {
-
-		> .wp-block-button,
-		&.is-content-justification-right > .wp-block-button, {
-			// Added (duplicate) specificity needed to override the default button margin.
-			&.wp-block-button {
-				margin-right: 0;
-				margin-left: 0;
-			}
+@supports ( column-gap:  var(--wp--style--block-gap, $blocks-block__margin)  ) {
+	.wp-block-buttons > .wp-block-button {
+		&.wp-block-button__width-25 {
+			width: calc(25% - (var(--wp--style--block-gap, $blocks-block__margin) * 0.75));
 		}
 
-		> .wp-block-button {
-			&.wp-block-button__width-25 {
-				width: calc(25% - #{ $blocks-block__margin * 0.75 });
-			}
-
-			&.wp-block-button__width-50 {
-				width: calc(50% - #{ $blocks-block__margin * 0.5 });
-			}
+		&.wp-block-button__width-50 {
+			width: calc(50% - (var(--wp--style--block-gap, $blocks-block__margin) * 0.5));
+		}
 
-			&.wp-block-button__width-75 {
-				width: calc(75% - #{ $blocks-block__margin * 0.25 });
-			}
+		&.wp-block-button__width-75 {
+			width: calc(75% - (var(--wp--style--block-gap, $blocks-block__margin) * 0.25));
+		}
 
-			&.wp-block-button__width-100 {
-				width: auto;
-				flex-basis: 100%;
-			}
+		&.wp-block-button__width-100 {
+			width: auto;
```

Comment:
I know this isn't a change in your PR, but I noticed while testing with TT1-blocks that when using the vertical variation, setting to 100% via `width: auto` results in the block being only as wide as its contents. Should this value be `width: 100%` instead? It appears to fix it for me in TT1-blocks.

![buttons-vertical-100-sml](https://user-images.githubusercontent.com/14988353/132634090-5b815f01-2834-4f07-8862-12a842268abe.gif)

Reply:
Oh, really good catch! Seems like this is only happening for vertically oriented _and_ center-aligned buttons, but that doesn't look good at all! Adding a fix in, thank you.